### PR TITLE
streaming: refactor diff/commit search

### PR DESCRIPTION
**Pull request by stefanhengl**
This refactor simplifies diff/commit search and at the same time aligns
its structure with that of the other search types.

The essence of the change is not to return a channel from
`searchCommitsInRepoStream` but instead take a `Sender` and
return a simple error.

### CODE REVIEW

File: cmd/frontend/graphqlbackend/search_commits.go [hunk 1]
```diff
@@ -351,39 +305,54 @@ func doSearchCommitsInRepoStream(ctx context.Context, db dbutil.DB, op search.Co
...
-				TimedOut: timedOut,
-			}
 		}
+
+		var searchErr error
+		if err != nil {
+			searchErr = err
+		} else {
+			searchErr = event.Error
+		}
```
Comment (stefanhengl) [inline cmd/frontend/graphqlbackend/search_commits.go:324]: This is pretty janky, but I wil get rid of it in a follow-up PR. Right now `logCommitSearchResultsToResolvers` returns an error but looking at the function the error is allways nil. So we can replace this block with
```
searchErr = event.Error
```

File: cmd/frontend/graphqlbackend/search_commits.go [hunk 2]
```diff
@@ -568,88 +530,36 @@ func searchCommitsInRepos(ctx context.Context, db dbutil.DB, args *search.TextPa
...
-		return
+		return searchCommitsInRepoStream(ctx, db, commitParams, params.ResultChannel)
 	}
 
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
-
-	var wg sync.WaitGroup
-	var mu sync.Mutex
+	g, ctx := errgroup.WithContext(ctx)
```
Comment (camdencheek) [inline cmd/frontend/graphqlbackend/search_commits.go:539]: <3 `errgroup`

File: cmd/frontend/graphqlbackend/search_commits.go [hunk 3]
```diff
@@ -568,88 +530,36 @@ func searchCommitsInRepos(ctx context.Context, db dbutil.DB, args *search.TextPa
...
-			defer mu.Unlock()
-			repoCommon, fatalErr := handleRepoSearchResult(repoRev, repoLimitHit, repoTimedOut, searchErr)
-			if fatalErr != nil {
-				err = errors.Wrapf(searchErr, "failed to search commit %s %s", params.ErrorName, repoRev.String())
-				cancel()
-			}
-			params.ResultChannel.Send(SearchEvent{
-				Stats: repoCommon,
+		rr := repoRev
+		g.Go(
```
Comment (camdencheek) [inline cmd/frontend/graphqlbackend/search_commits.go:547]: Some weird indentation here

**Review by camdencheek — APPROVED** [review]

File: cmd/frontend/graphqlbackend/search_commits.go [hunk 4]
```diff
@@ -351,44 +305,54 @@ func doSearchCommitsInRepoStream(ctx context.Context, db dbutil.DB, op search.Co
 		repoName = *op.RepoRevs.Repo
 	}
 
+	var results []*CommitSearchResultResolver
+	var stats streaming.Stats
```
[the remark this answers is not in the document]
Comment (stefanhengl) [reply cmd/frontend/graphqlbackend/search_commits.go]: I removed it.

**Pull request merged**